### PR TITLE
Update for daml 2.1.0

### LINKED ISSUE #12: Edit Java bindings Ping Pong examples to work with Daml 2.0

### PR DESCRIPTION
**Pull request by realvictorprm**
Fixes #12

### CODE REVIEW

File: src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java [hunk 1]
```diff
@@ -19,12 +19,16 @@
...
+import com.daml.ledger.api.v1.admin.UserManagementServiceGrpc.UserManagementServiceBlockingStub;
+import com.daml.ledger.api.v1.admin.UserManagementServiceOuterClass.GetUserRequest;
+import com.daml.ledger.api.v1.admin.UserManagementServiceOuterClass.GetUserResponse;
 import com.daml.ledger.api.v1.ValueOuterClass.Identifier;
 import com.daml.ledger.api.v1.ValueOuterClass.Record;
 import com.daml.ledger.api.v1.ValueOuterClass.RecordField;
 import com.daml.ledger.api.v1.ValueOuterClass.Value;
-import com.digitalasset.daml_lf_1_8.DamlLf;
-import com.digitalasset.daml_lf_1_8.DamlLf1;
+import com.daml.daml_lf_1_14.DamlLf;
```
Comment (cocreature) [inline src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java]: can you drop `detectPingPongPackageId` and therefore also the dep on this? it looks like you get the package id via the cli arg now.

Comment (realvictorprm) [reply src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java]: yeah I noticed that one too and did that already just now locally :D

File: src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java [hunk 2]
```diff
@@ -61,9 +55,16 @@ public static void main(String[] args) {
...
-        // this is helpful during development when the package id changes a lot due to likely frequent changes to the DAML code
-        String packageId = detectPingPongPackageId(channel, ledgerId);
+        // fetch the party IDs that got created in the Daml init script
+        String aliceParty = fetchPartyId(channel, ALICE_USER);
+        String bobParty = fetchPartyId(channel, BOB_USER);
+
+        // inspect the packages on the ledger and extract the package id of the package
+        // containing the PingPong module
+        // this is helpful during development when the package id changes a lot due to
+        // likely frequent changes to the DAML code
```
Comment (cocreature) [inline src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java]: comment seems outdated.

File: src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java [hunk 3]
```diff
@@ -61,9 +55,12 @@ public static void main(String[] args) {
         // fetch the ledger ID, which is used in subsequent requests sent to the ledger
         String ledgerId = fetchLedgerId(channel);
 
-        // inspect the packages on the ledger and extract the package id of the package containing the PingPong module
-        // this is helpful during development when the package id changes a lot due to likely frequent changes to the DAML code
-        String packageId = detectPingPongPackageId(channel, ledgerId);
+        // fetch the party IDs that got created in the Daml init script
```
Comment (cocreature) [inline src/main/java/examples/pingpong/grpc/PingPongGrpcMain.java:58]: i would have passed those in as arguments and resolve them outside but don’t feel strongly about this.

File: src/main/java/examples/pingpong/reactive/PingPongReactiveMain.java [hunk 4]
```diff
@@ -50,130 +49,81 @@ public static void main(String[] args) {
...
-        Identifier pongIdentifier = new Identifier(packageId, "PingPong", "Pong");
-
+        var userManagementClient = client.getUserManagementClient();
+        String aliceParty = userManagementClient.getUser(new GetUserRequest(ALICE_USER)).blockingGet().getUser().getPrimaryParty().get();
+        String bobParty = userManagementClient.getUser(new GetUserRequest(BOB_USER)).blockingGet().getUser().getPrimaryParty().get();
+
+        // inspect the packages on the ledger and extract the package id of the package
+        // containing the PingPong module
+        // this is helpful during development when the package id changes a lot due to
+        // frequent changes to the DAML code
```
Comment (cocreature) [inline src/main/java/examples/pingpong/reactive/PingPongReactiveMain.java]: outdated comment

File: src/main/java/examples/pingpong/reactive/PingPongReactiveMain.java [hunk 5]
```diff
@@ -50,130 +49,81 @@ public static void main(String[] args) {
...
-                            new DamlRecord.Field("sender", new Party(sender)),
-                            new DamlRecord.Field("receiver", new Party(receiver)),
-                            new DamlRecord.Field("count", new Int64(0))
-                    )
-            );
+            // command that creates the initial Ping contract with the required parameters
+            // according to the model
+
+            Command createCommand = Command.newBuilder().setCreate(
+                    CreateCommand.newBuilder()
```
Comment (cocreature) [inline src/main/java/examples/pingpong/reactive/PingPongReactiveMain.java:100]: indentation seems a bit weird, not sure if this is autoformatted, maybe with 8-space indents?

File: start.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+echo "Compiling code"
+mvn compile
+
+# Could also run this manually in another terminal without the redirects
+echo "Starting sandbox"
+daml start --start-navigator false --sandbox-port 7600 > sandbox.log 2>&1 & PID=$!
+
+
+while [[ "$(getSandboxPid)" -eq '' ]]
```
Comment (cocreature) [inline start.sh:28]: I recommend that you wait for the port file to be written out not for the port to be open. the latter can be somewhat unreliable.

**Review by cocreature — DISMISSED** [review]
looks good with some minor comments

File: src/main/java/examples/pingpong/reactive/PingPongReactiveMain.java (hunk 5 above)
Comment (realvictorprm) [reply src/main/java/examples/pingpong/reactive/PingPongReactiveMain.java:100]: good point... I was thinking about adding automatic reformatting to the project and do a reformat once (with as less changes as possible).

File: start.sh (hunk 6 above)
Comment (realvictorprm) [reply start.sh:28]: I think this should be fine for now as this is not something super critical and so far it worked out for me fine :)

**Review by cocreature — APPROVED** [review]

**Pull request merged**